kotlin/src/main/kotlin/redux/api/Reducer.kt

```diff
@@ -0,0 +1,58 @@
+package redux.api
+
+/*
+ * Copyright (C) 2016 Michael Pardo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A reducer accepts an accumulation and a value and returns a new accumulation. They are used to reduce a collection of
+ * values down to a single value.
+ *
+ * @see <a href="http://redux.js.org/docs/basics/Reducers.html">http://redux.js.org/docs/basics/Reducers.html</a>
+ */
+interface Reducer<S : Any> {
```

Comment:
Reducer does not provide a `combine`method because it is considered as logic (out of the scope of the API).